src/com/lftechnology/java/training/niraj/ems/services/CrudService.java

```diff
@@ -0,0 +1,34 @@
+package com.lftechnology.java.training.niraj.ems.services;
+
+public interface CrudService<T, S> {
+
+    /**
+     * Saves information
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param t
+     * @return
+     * @throws Exception
+     */
+    public T save(T t) throws Exception;
+
+    /**
+     * Finds a record
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param s
+     * @return
+     * @throws Exception
+     */
+    public T find(S s) throws Exception;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)